(feature) optional param added to throwIfNotFound method

vasdhara opened the pull request:
closes #1722 

Is this what was required for the feature request? Please let me know if I have missed anything. A first-time contributor so slightly overwhelmed.

koskimas commented on doc/api/query-builder/other-methods.md:
```diff
@@ -1071,13 +1071,19 @@ console.log(firstPerson.age);
+
+| Argument | Type   | Description                      |
+| -------- | ------ | -------------------------------- |
+| data     | object | optional object with custom data |
```
There's no way for people to use this based on this documentation. Where does this data go? What properties can it have? How are different properties treated (like the special case for `message`).

vasdhara replied:
I've updated the description and added a custom type example, is that sufficient?

koskimas replied:
That's excellent!

koskimas commented on tests/integration/find.js:
```diff
@@ -574,7 +574,7 @@ module.exports = session => {
         });
 
-        it('.throwIfNotFound() with empty result', done => {
+        it('custom .throwIfNotFound() with empty result', done => {
```
Why did you add the word `custom` here?

vasdhara replied:
was a typo, reverted

koskimas commented:
Hi @vasdhara! Thank you so much for your contribution! I added a couple of comments. You also need to update the typescript typings.

vasdhara commented:
@koskimas thanks. added fixes. Also, added a new interface for throwIfNotFound typings, since I couldn't find a previously written interface that could be reused in this case.
